upgrade/steps.go

```diff
@@ -182,14 +179,19 @@ func UpgradeProviderVersion(
 		// instead of using version tags because we can't ensure that the upstream is
 		// versioning their go modules correctly.
 		//
-		// It they are versioning correctly, `go mod tidy` will resolve the SHA to a tag.
+		// If they are versioning correctly, `go mod tidy` will resolve the SHA to a tag.
 		steps = append(steps,
 			step.F("Lookup Tag SHA", func(context.Context) (string, error) {
-				path, err := getGitHubPath(goMod.Upstream.Path)
-				if err != nil {
-					return "", err
+				upstreamOrg := GetContext(ctx).UpstreamProviderOrg
+				upstreamRepo := GetContext(ctx).UpstreamProviderName
+				gitHostPath := "https://github.com/" + upstreamOrg + "/" + upstreamRepo
+
+				// special case: we need to use the GitLab url for getting git refs.
+				if upstreamOrg == "terraform-provider-gitlab" {
+					gitHostPath = "https://gitlab.com/gitlab-org/terraform-provider-gitlab"
 				}
-				refs, err := gitRefsOf(ctx, "https://"+modPathWithoutVersion(path),
+
+				refs, err := gitRefsOf(ctx, gitHostPath,
 					"tags")
```

Comment:
The line is now pretty short (within lint bounds), so let's put ` "tags")` back on the same line.
```suggestion
				refs, err := gitRefsOf(ctx, gitHostPath, "tags")
```